### PR TITLE
Fix #625

### LINKED ISSUE #625: A player that is out cannot collide with other players
### Context

While a player is out, as a result of being killed, it should not be able to run in collisions with other players, instead passing through them as if it didn't exist.

### Test
- [x] Test is ready and on develop 

```cpp
#define FIX_ISSUE_625
{
      // (625) A player that is out cannot collide with other players
    game g;
    const coordinate c_p2 = g.get_player(1).get_position();
    g.get_player(0).place_to_position(c_p2);
    assert(are_colliding(g.get_player(0), g.get_player(1)));
    g.kill_player(0);
    assert(!are_colliding(g.get_player(0), g.get_player(1)));
}
#endif // FIX_ISSUE_XXX
```

### PR DESCRIPTION
**Pull request by Luxwor**
Me and Claudio fixed #625 :)

### CODE REVIEW

File: src/game.cpp [hunk 1]
```diff
@@ -996,10 +996,10 @@ void test_game() //!OCLINT tests may be many
     assert(is_dead(p));
   }
 #endif
-
+  #define FIX_ISSUE_625
```
Comment (richelbilderbeek) [inline src/game.cpp:999]: Well done, you can remove the `#defines` after fixing an Issue :-)

File: src/player.cpp [hunk 2]
```diff
@@ -129,11 +129,16 @@ void add_action(player& p, action_type action) noexcept
 
 bool are_colliding(const player &lhs, const player &rhs) noexcept
 {
+  if (lhs.get_state() == player_state::active &&
+      rhs.get_state() == player_state::active)
+  {
```
Comment (richelbilderbeek) [inline src/player.cpp:134]: In C++ it is common to do

```c++
if (a == b
  && c == d
  && e == f
)
{
  // Stuff
}
```

Notice the place for the `&&`: it's at the **start** at each line (note in R one cannot do this)

**Review by richelbilderbeek — CHANGES_REQUESTED** [review]
Nearly there :-)

File: src/player.cpp (hunk 2 above)
Comment (Luxwor) [reply src/player.cpp:134]: did it!

**Review by richelbilderbeek — APPROVED** [review]
Brilliant!

**Pull request merged**